File: detect_color.py

try:
    from PIL import Image
    from collections import Counter
    
    img = Image.open("public/images/logo-onette.png")
    img = img.convert("RGB")
    width, height = img.size
    
    # Sample pixels (every 10th pixel to be fast)
    pixels = []
    for x in range(0, width, 10):
        for y in range(0, height, 10):
            pixels.append(img.getpixel((x, y)))
            
    most_common = Counter(pixels).most_common(1)[0][0]
    hex_color = '#{:02x}{:02x}{:02x}'.format(most_common[0], most_common[1], most_common[2])
    print(f"Most common color: {hex_color}")
    
except ImportError:
    print("PIL not installed")
except Exception as e:
    print(f"Error: {e}")

File: detect_color.js

import sharp from 'sharp';

async function checkCorners() {
    try {
        const image = sharp('public/images/logo-onette.png');
        const { data, info } = await image.raw().toBuffer({ resolveWithObject: true });

        const getPixel = (x, y) => {
            const i = (y * info.width + x) * info.channels;
            const r = data[i];
            const g = data[i + 1];
            const b = data[i + 2];
            return '#' + ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
        };

        console.log(`Top-Left: ${getPixel(0, 0)}`);
        console.log(`Top-Right: ${getPixel(info.width - 1, 0)}`);
        console.log(`Bottom-Left: ${getPixel(0, info.height - 1)}`);
        console.log(`Bottom-Right: ${getPixel(info.width - 1, info.height - 1)}`);

    } catch (e) {
        console.error(e);
    }
}

checkCorners();
